collect_app/src/test/java/org/odk/collect/android/preferences/ChangeAdminPasswordDialogTest.java

```diff
@@ -0,0 +1,166 @@
+package org.odk.collect.android.preferences;
+
+import android.content.DialogInterface;
+import android.content.res.Configuration;
+import android.text.InputType;
+import android.widget.CheckBox;
+import android.widget.EditText;
+
+import androidx.appcompat.app.AlertDialog;
+import androidx.fragment.app.FragmentActivity;
+import androidx.fragment.app.FragmentManager;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.odk.collect.android.R;
+import org.odk.collect.android.support.RobolectricHelpers;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.android.controller.ActivityController;
+import org.robolectric.shadows.ShadowDialog;
+
+import static android.view.View.VISIBLE;
+import static junit.framework.TestCase.assertTrue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertFalse;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.robolectric.Shadows.shadowOf;
+
+@RunWith(RobolectricTestRunner.class)
+public class ChangeAdminPasswordDialogTest {
+
+    private ActivityController<FragmentActivity> activity;
+    private FragmentManager fragmentManager;
+    private ChangeAdminPasswordDialog dialogFragment;
+    private ChangeAdminPasswordDialog.ChangePasswordDialogCallback callback;
+
+    @Before
+    public void setup() {
+        activity = RobolectricHelpers.buildThemedActivity(FragmentActivity.class);
+        activity.setup();
+
+        fragmentManager = activity.get().getSupportFragmentManager();
+        dialogFragment = new ChangeAdminPasswordDialog();
+
+        callback = mock(ChangeAdminPasswordDialog.ChangePasswordDialogCallback.class);
+        dialogFragment.callback = callback;
+    }
+
+    @Test
+    public void dialogIsCancellable() {
+        dialogFragment.show(fragmentManager, "TAG");
+        assertThat(shadowOf(dialogFragment.getDialog()).isCancelable(), equalTo(true));
+    }
+
+    @Test
+    public void shouldShowCorrectButtons() {
+        dialogFragment.show(fragmentManager, "TAG");
+        AlertDialog dialog = (AlertDialog) ShadowDialog.getLatestDialog();
+
+        assertThat(dialog.getButton(DialogInterface.BUTTON_POSITIVE).getVisibility(), equalTo(VISIBLE));
+        assertThat(dialog.getButton(DialogInterface.BUTTON_POSITIVE).getText(),
+                equalTo(activity.get().getString(R.string.ok)));
+        assertThat(dialog.getButton(DialogInterface.BUTTON_NEGATIVE).getVisibility(), equalTo(VISIBLE));
+        assertThat(dialog.getButton(DialogInterface.BUTTON_NEGATIVE).getText(),
+                equalTo(activity.get().getString(R.string.cancel)));
+    }
+
```

Comment:
Hi @seadowg! I'm not sure if the above two tests are necessary.

Reply:
The first one `dialogIsCancellable` checks that the back button would works so I think that's good. The second seems like it isn't needed as you  interact with all the elements in the tests below.

Reply:
Thanks @seadowg! So can I remove the second test case, for all my subsequent PRs on dialog disappearing issue. As for all of them, somewhere later in some other test, I have interacted with the elements, so I think, this part of code is redundant.